lib/routemaster/fetcher.rb

```diff
@@ -34,12 +28,30 @@ def initialize(host)
     # and `body`. The body is a `Hashie::Mash` if the response was JSON, a
     # string otherwise.
     def get(url, params:nil, headers:nil)
-      r = _connection.get(url, params, headers)
+      r = _connection_for(url).get(url, params, headers)
       Hashie::Mash.new(status: r.status, headers: r.headers, body: r.body)
     end
 
+    def use(*args, &block)
```

Comment:
This API is leaky, since it allows changing the middleware stack - which is actually frozen the first time a connection of established (since we're memoising the connection).

If you really insist on using block-based configuration sugar, you could do something like:

```
# build the connection object earlier
def Fetcher.initialize
  @conn = Faraday.new do |f|
    # our stuff ...
    yield f
  end
end

Fetcher.new do |faraday|
  faraday.use MyMiddleware
end
```

But just as simply, you could:

```
class Fetcher 
  def initialize(middleware: [])
    @middleware = middleware
  end

  def _connection
     @_connection ||= Faraday.new do |f|
       # ...
       @middleware.each { |m| f.use m }
     end
  end


```